span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToLogRecordsTransformer.java

```diff
@@ -0,0 +1,92 @@
+package org.hypertrace.core.spannormalizer.jaeger;
+
+import static org.hypertrace.core.spannormalizer.constants.SpanNormalizerConstants.SPAN_NORMALIZER_JOB_CONFIG;
+
+import com.google.protobuf.util.Timestamps;
+import com.typesafe.config.Config;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.Span;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.hypertrace.core.datamodel.Attributes;
+import org.hypertrace.core.datamodel.LogEvent;
+import org.hypertrace.core.datamodel.LogEvents;
+import org.hypertrace.core.spannormalizer.util.AttributeValueCreator;
+
+public class JaegerSpanToLogRecordsTransformer
+    implements Transformer<byte[], Span, KeyValue<String, LogEvents>> {
```

Comment:
So we're implementing this as a completely parallel transformer? I mostly like that, but it does cause a few issues (noted below)

Reply:
Yeah, I found this to be clean. Otherwise, I will have to use the low level kafka streams processor api and manually forward to respective topics.